mail/pfSense-pkg-postfix/files/usr/local/pkg/postfix.inc

```diff
@@ -0,0 +1,990 @@
+					break;
+				case '01hour':
+					$s_minute = "0";
+					$cron_postfix_sqlite = $cron_cmd_sqlite . " 01hour";
+					break;
+				case '24hours':
+					$s_minute = "0";
+					$s_hour = "0";
+					$cron_postfix_sqlite = $cron_cmd_sqlite . " 24hours";
+					break;
+			}
+		}
+	}
+	
+	// update cron
+	if ($postfix_enabled == "on") {
+		if ($postfix_recipients_config['enable_ldap'] || $postfix_recipients_config['enable_url']) {
+			install_cron_job ("{$cron_cmd_recipients}", true, $r_minute, $r_hour, $r_mday, $r_month, $r_wday, $r_who);
+		} else {
+			install_cron_job ("{$cron_cmd_recipients}", false);
+		}
+		if ($cron_sqlite_queue != "" && $cron_sqlite_queue != "never") {
+			// First remove the previous schedule since the command was appended as well
+			install_cron_job ("{$cron_cmd_sqlite}", false);
+			install_cron_job ("{$cron_postfix_sqlite}", true, $s_minute, $s_hour, $s_mday, $s_month, $s_wday, $s_who);
+		} else {
+			install_cron_job ("{$cron_cmd_sqlite}", false);
+		}
+	} else {
+		install_cron_job ("{$cron_cmd_recipients}", false);
+		install_cron_job("{$cron_cmd_sqlite}", false);
+	}
+}
+
+function sync_package_postfix($via_rpc = "no") {
+	global $g, $config;
+
+	log_error("sync_package_postfix called with via_rpc={$via_rpc}");
+	# detect boot process
+	if (is_array($_POST)) {
+		if (function_exists("platform_booting")) {
+			if (!platform_booting()) {
+				unset($boot_process);
+			} else {
+				$boot_process = "on";
+			}
+		} elseif (!($g['booting'])) {
+			unset($boot_process);
+		} else {
+			$boot_process = "on";
+		}
+	}
+
+	if (is_process_running("master") && isset($boot_process) && $via_rpc == "no") {
+		return;
+	}
+
+	// check patch in /etc/inc/config.
+	$relay_domains = "";
+	$transport = "";
+	$postfix_config=$config['installedpackages']['postfix']['config'][0];
+	if (is_array($config['installedpackages']['postfixdomains'])) {
+		$postfix_domains=$config['installedpackages']['postfixdomains']['config'][0];
+	}
+	$message_size_limit = ($postfix_config['message_size_limit'] ? $postfix_config['message_size_limit'] : "10240000");
+	$process_limit = ($postfix_config['process_limit'] ? $postfix_config['process_limit'] : "100");
+	if (is_array($postfix_domains['row'])) {
+		foreach ($postfix_domains['row'] as $postfix_row) {
+			$relay_domains .= ' ' . $postfix_row['domain'];
+			if (!empty($postfix_row['mailserverip'])) {
+				$transport .= $postfix_row['domain'] . " smtp:[" . $postfix_row['mailserverip'] . "]\n";
+			}
+		}
+	}
+	
+	// check cron
+	check_cron();
+	
+	// check logging
+	if ($postfix_config['log_to']) {
+			switch($postfix_config['log_to']) {
+				case 'maillog':
+					system("/usr/bin/touch /var/log/maillog");
+					$mail_syslog = "mail.crit;";
+				break;
+				case 'none':
+					$mail_syslog = "mail.crit;";
+				break;
+				default:
+					$mail_syslog = 'mail.*;';
+				break;
+			}
+            #update /etc/inc/system.inc
+            $sys_log_file = '/etc/inc/system.inc';
```

Comment:
Packages will never be allowed to touch system files anymore. It's wrong and dangerous